src/coreclr/nativeaot/Runtime/inc/CommonTypes.h

```diff
@@ -6,9 +6,30 @@
 
 #include <cstddef>
 #include <cstdint>
-#include <cstdlib>
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifdef TARGET_WINDOWS
+
 #include <new>
 
+#else
+
+extern "C" void __cxa_pure_virtual();
+
+namespace std
```

Comment:
There can be a collision with C++ library that the user may want to link statically. I think it would be best to avoid stomping on or reimplementing anything that C++ runtime library provides, prefer refactoring the code to avoid the dependencies C++ runtime library.

Reply:
Changed `std` to `nativeaot`.